attributes/default.rb

```diff
@@ -498,6 +498,7 @@
 default['cluster']['raid_vol_ids'] = nil
 default['cluster']['dns_domain'] = nil
 default['cluster']['use_private_hostname'] = 'false'
+default['cluster']['workaround_with_private_hostname'] = 'false'
```

Comment:
I don't like the name "workaround" for a variable.
Let's rename to "private_hostnames_in_hosts_file" or something explaining what is doing/enabling that variable.

Reply:
yes another potential name as suggested in https://github.com/aws/aws-parallelcluster/pull/3644#pullrequestreview-849229268 could be `enable_node_hostnames_local_resolution`. Or also `register_node_hostnames_in_hosts_file`/`node_hostnames_in_hosts_file`

Reply:
Done by using the name `add_node_hostnames_in_hosts_file`